src/config.rs

```diff
@@ -37,6 +60,21 @@ impl Display for OverrideReason {
     }
 }
 
+#[derive(Debug, Default)]
+pub struct OverrideCfg {
```

Comment:
If we make channel optional as per the above, then this struct becomes less necessary perhaps?  Otherwise if you want to ensure that there remains API separation between the `rust-toolchain` format and the internal structs used, then this should probably contain the `channel: Option<Toolchain<'_>>` too.

Reply:
I think decoupling them is a good idea as you mention. Also, I've realized that this struct does not need to be exposed outside the `config` module, so I kept the previous API of `find_override` returning the toolchain and the override reason.

Reply:
Sounds good.